api/client/client.go

```diff
@@ -111,19 +145,30 @@ type Config struct {
 	// KeepAliveCount specifies the amount of missed keep alives
 	// to wait for before declaring the connection as broken
 	KeepAliveCount int
-	// TLS is the client's TLS config
-	TLS *tls.Config
+	// CredentialProviders are used to authenticate the client's connection to the server.
+	// The first provider to provide valid credentials will be used to dial the server.
+	CredentialsProviders []CredentialsProvider
+	// Credentials is the client's current Credentials for authentication. If this is provided
+	// in client creation, it will be used to create a basic CredentialProvider.
+	Credentials *Credentials
+	// NoPingCheck disables the ping check on client initialization. The ping check is used
+	// to verify and authenticate the connection, but in cases where you don't immediately
+	// expect the connection to be usable, use NoPingCheck to disable this check.
+	NoPingCheck bool
```

Comment:
Is there a reason to make this via configuration instead of deciding on the caller's side whether to do a ping? I'd remove this.

Reply:
The intended functionality is for the client to automatically Ping to 1) check that the connection is open and authenticated, in order to choose the right credentials from the list of credential loaders and 2) check the version of the server and compare it to the client (not yet implemented). `NoPingCheck` is primarily for backwards compatibility with the `lib/client.Client`, so I'll add a sentence to the comment describing that.